wolfcrypt/src/chacha.c

```diff
@@ -361,6 +429,13 @@ int wc_Chacha_Process(ChaCha* ctx, byte* output, const byte* input,
     return 0;
 }
 
+void wc_Chacha_purge_current_block(ChaCha* ctx) {
+    if (ctx->left > 0) {
+        byte scratch[CHACHA_CHUNK_BYTES] = {};
+        (void)wc_Chacha_Process(ctx, scratch, scratch, CHACHA_CHUNK_BYTES - ctx->left);
```

Comment:
Is this really the best way to cleanup the leftovers? `ctx->left` and `ctx->over`?

Reply:
Yeah, best I can tell.  Note that the semantics of `ChaCha.X[CHACHA_MATRIX_CNT_IV]` are different for C vs asm, hence the apparently frivolous call to `wc_Chacha_Process()`.  In a previous try I incremented the counter directly from wc_Chacha_purge_current_block(), which in due course led to much sadness and the current solution.